src/oc/storage/async/notification.clj

```diff
@@ -75,14 +76,23 @@
   (timbre/trace "Notification request:" trigger)
   (schema/validate NotificationTrigger trigger)
   (timbre/info "Sending request to topic:" config/aws-sns-storage-topic-arn)
-  (sns/publish
-    {:access-key config/aws-access-key-id
-     :secret-key config/aws-secret-access-key}
-     :topic-arn config/aws-sns-storage-topic-arn
-     :subject (str (name (:notification-type trigger))
-                   " on " (name (:resource-type trigger))
-                   ": " (-> trigger :current :uuid))
-     :message (json/generate-string trigger {:pretty true}))
+  (let [subject (str (name (:notification-type trigger))
+                     " on " (name (:resource-type trigger))
+                     ": " (-> trigger :current :uuid))]
+    (try
+      (sns/publish
+       {:access-key config/aws-access-key-id
+        :secret-key config/aws-secret-access-key}
+       :topic-arn config/aws-sns-storage-topic-arn
+       :subject subject
+       :message (json/generate-string trigger {:pretty true}))
```

Comment:
@thepug maybe `(json/generate-string trigger {:pretty true})` can go into a var in the let? it's called again if sns fails

Reply:
updated